### ISSUE TITLE
Disallow optional parameters notation with question mark

### ISSUE THREAD

**Issue by yvele** [gajus/eslint-plugin-jsdoc#489]
With **auto fix**

I would like to disallow optional parameters notation with question mark character `?`.

Disallow this:

```js
/**
 * @param {string?} foo Optional parameter.
 */
```

and this:

```js
/**
 * @param {?string} foo Optional parameter.
 */
```

In favor of this (auto fixed):

```js
/**
 * @param {string} [foo] Optional parameter.
 */
```

Actually both `{string?}` and `{?string}` cannot be disallowed and leads to inconstancy when writing optional parameter notation.

Also I'm not sure about the `{string} [foo]` notation, is this really the "official" notation to use when writing optional parameters?

I think so 🤔 https://jsdoc.app/tags-param.html#optional-parameters-and-default-values

> An optional parameter (using JSDoc syntax)
> 
> ```js
> /**
>  * @param {string} [somebody] - Somebody's name.
>  */
> function sayHello(somebody) {
>     if (!somebody) {
>         somebody = 'John Doe';
>     }
>     alert('Hello ' + somebody);
> }
> ```

**Comment by l1bbcsg** [comment]
I believe you are confusing _optional_ and _nullable_ types. `?string` is nullable and essentially an equivalent of `(string|null)` both in [vanilla](https://jsdoc.app/tags-type.html) and [GCC](https://github.com/google/closure-compiler/wiki/Types-in-the-Closure-Type-System). 

`@param {?string} foo` is a perfectly valid tag which means function parameter `foo` is mandatory and could be either `string` or `null`.

Optional parameter would be either `@param {string=} foo` or `@param {string} [foo]` and roughly equivalent of `@param {string|undefined} foo`. ECMAScript Default Parameters are fine too.

**Comment by yvele** [comment]
Oh yeah right, thank you!

**Comment by brettz9** [comment]
FWIW: https://github.com/jsdoc/jsdoc/issues/1532

**Comment by yvele** [comment]
And is there a way to disallow _nullable_ types? I'm pretty sure my team doesn't have any `?string` nullable use-case 🤷‍♂
